src/main/java/it/gov/pagopa/reward/connector/repository/HpanInitiativesAtomicOpsRepositoryImpl.java

```diff
@@ -20,9 +22,12 @@ public class HpanInitiativesAtomicOpsRepositoryImpl implements HpanInitiativesAt
     public HpanInitiativesAtomicOpsRepositoryImpl(ReactiveMongoTemplate mongoTemplate) {
         this.mongoTemplate = mongoTemplate;
     }
-
     @Override
     public Mono<UpdateResult> createIfNotExist(HpanInitiatives hpanInitiatives) {
+      return MongoRequestRateTooLargeRetryer.withRetry(createIfNotExistInner(hpanInitiatives), MongoRequestRateTooLargeUtilities.maxRetry, MongoRequestRateTooLargeUtilities.maxMillisElapsed);
+    }
+
+    public Mono<UpdateResult> createIfNotExistInner(HpanInitiatives hpanInitiatives) {
```

Comment:
inner method should be private
```suggestion
    private Mono<UpdateResult> createIfNotExistInner(HpanInitiatives hpanInitiatives) {
```